include/sound/sof/dai.h

```diff
@@ -50,6 +50,10 @@
 #define SOF_DAI_FMT_INV_MASK		0x0f00
 #define SOF_DAI_FMT_CLOCK_PROVIDER_MASK	0xf000
 
+/* DAI_CONFIG flags */
+#define SOF_DAI_CONFIG_FLAGS_HW_PARAMS	(1 << 0) /**< DAI_CONFIG sent during hw_params stage */
+#define SOF_DAI_CONFIG_FLAGS_HW_FREE	(1 << 1) /**< DAI_CONFIG sent during hw_free stage */
```

Comment:
We seem to have 3 valid cases: both flags 0, and each of the flags is 1 while the other one is 0. I.e. allowed combinations are 00, 01, 10. Shouldn't we then define this as a 2-bit field, define 3 macros for the values and use `set_mask_bits()` to set both bits (atomically)

Reply:
I agree, this would be cleaner.

Reply:
I am not opposed to the suggestion, I was erring more on the side of future proofing. 

Yes we only have three configurations today but this DAI_CONFIG is a nightmare and it's a matter of time before we need extensions... 

I hesitated to e.g. add HW_PARAMS and PREPARE since we send the DAI_CONFIG at different times for different DAIs. We also send a DAI_CONFIG during the TRIGGER_STOP stage for HDaudio so that's 5 configurations...

Reply:
*effectively* it's now doing what I proposed, but "formally" I think it should be defined a bit differently now. Since we now have a 2-bit field at shift 0, we have to define `SOF_DAI_CONFIG_FLAGS_HW_FREE` not as `1 << 1` but as `2 << 0`, which of course produce the same result :-) But having a 2-bit field at offset 0 means that our choices now are `0 << 0`, `1 << 0`, `2 << 0` and `3 << 0` where the latter isn't used yet

Reply:
followed @lyakh 's suggestion